spot-contracts/contracts/FeePolicy.sol

```diff
@@ -307,23 +299,55 @@ contract FeePolicy is IFeePolicy, OwnableUpgradeable {
     /// @inheritdoc IFeePolicy
     /// @dev Swapping by minting perps reduces system dr, i.e) drPost < drPre.
     function computeUnderlyingToPerpVaultSwapFeePerc(
-        uint256 /*drPre*/,
+        uint256 drPre,
         uint256 drPost
     ) external view override returns (uint256) {
         // When the after op deviation ratio is below the bound,
         // swapping is disabled. (fees are set to 100%)
-        return (drPost < deviationRatioBoundLower ? ONE : vaultUnderlyingToPerpSwapFeePerc);
+        if (drPost <= drHardBounds.lower) {
+            return ONE;
+        }
+        return
+            LineHelpers
+                .computePiecewiseAvgY(
+                    Line({ x1: drHardBounds.lower, y1: flashMintFeePercs.upper, x2: ONE, y2: flashMintFeePercs.lower }),
+                    Line({ x1: ONE, y1: flashMintFeePercs.lower, x2: drHardBounds.upper, y2: flashMintFeePercs.lower }),
+                    Range({ lower: drPost, upper: drPre }),
+                    ONE
+                )
+                .toUint256();
```

Comment:
Should we have this function just return a uint, like in the original version?

Reply:
you mean `computePiecewiseAvgY`? I think that always returned an integer 

https://github.com/ampleforth/spot/blob/development/spot-vaults/contracts/_utils/LineHelpers.sol